pkg/utils/helpers.go

```diff
@@ -282,3 +282,36 @@ func PatchAddonCondition(ctx context.Context, addonClient addonv1alpha1client.In
 		ctx, new.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status")
 	return err
 }
+
+// AddonManagementFilterFunc is to check if the addon should be managed by addon manager or self-managed
+type AddonManagementFilterFunc func(cma *addonapiv1alpha1.ClusterManagementAddOn) bool
+
+func ManagedByAddonManager(cma *addonapiv1alpha1.ClusterManagementAddOn) bool {
+	if len(cma.Annotations) == 0 {
+		return false
+	}
+
+	value, ok := cma.Annotations[addonapiv1alpha1.AddonLifecycleAnnotationKey]
+	if !ok {
+		return false
+	}
+
+	return value == addonapiv1alpha1.AddonLifecycleAddonManagerAnnotationValue
+}
+
+func ManagedBySelf(cma *addonapiv1alpha1.ClusterManagementAddOn) bool {
+	return !ManagedByAddonManager(cma)
```

Comment:
Do we need to limit the user can only define annotation value to `"self"` and `addon-manager`, or anything other than `addon-manager` will be treated as self?